src/components/SearchBox/SearchBox.tsx

```diff
@@ -1,107 +1,126 @@
-import React, { useRef, useState } from 'react';
-import styled from 'styled-components';
-import throttle from 'lodash/throttle';
-import SearchIcon from '@material-ui/icons/Search';
-import Paper from '@material-ui/core/Paper';
-import IconButton from '@material-ui/core/IconButton';
+import React, { useState, useEffect } from 'react';
 import Router from 'next/router';
-import { fetchJson } from '../../services/fetch';
+// headlessUI
+import { Combobox } from '@headlessui/react';
+// icons
+import { XMarkIcon, MagnifyingGlassIcon } from '@heroicons/react/24/solid';
+// context
 import { useMapStateContext } from '../utils/MapStateContext';
 import { useFeatureContext } from '../utils/FeatureContext';
-import { AutocompleteInput } from './AutocompleteInput';
+// helpers
 import { t } from '../../services/intl';
-import { ClosePanelButton } from '../utils/ClosePanelButton';
-import { isDesktop, useMobileMode } from '../helpers';
+import { useMobileMode } from '../helpers';
+import { fetchChoices } from './searchBoxUtils';
+import { onHighlightFactory, onSelectedFactory } from './onSelectedFactory';
+// components
+import SearchChoice from './SearchChoice';
 
-const TopPanel = styled.div`
-  position: absolute;
-  height: 72px;
-  box-shadow: 0 10px 20px 0 rgba(0, 0, 0, 0.12);
-  background-color: ${({ theme }) => theme.palette.background.searchBox};
-  padding: 10px;
-  box-sizing: border-box;
-
-  z-index: 1200;
-
-  width: 100%;
-  @media ${isDesktop} {
-    width: 410px;
-  }
-`;
-
-const StyledPaper = styled(Paper)`
-  padding: 2px 4px;
-  display: flex;
-  align-items: center;
-
-  .MuiAutocomplete-root {
-    flex: 1;
-  }
-`;
-
-const SearchIconButton = styled(IconButton)`
-  svg {
-    transform: scaleX(-1);
-    filter: FlipH;
-    -ms-filter: 'FlipH';
-  }
-`;
-
-const getApiUrl = (inputValue, view) => {
-  const [zoom, lat, lon] = view;
-  const lvl = Math.max(0, Math.min(16, Math.round(zoom)));
-  const q = encodeURIComponent(inputValue);
-  return `https://photon.komoot.io/api/?q=${q}&lon=${lon}&lat=${lat}&zoom=${lvl}`;
-};
-
-const fetchOptions = throttle(async (inputValue, view, setOptions) => {
-  const searchResponse = await fetchJson(getApiUrl(inputValue, view));
-  const options = searchResponse.features;
-  setOptions(options || []);
-}, 400);
+interface Choice {
+  id: string;
+  properties: {
+    name: string;
+  };
+}
 
 const SearchBox = () => {
```

Comment:
Regarding this component as whole, i created another PR https://github.com/zbycz/osmapp/pull/172. I will DM you about it as well.